src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -286,8 +273,17 @@ internal static string ValidateIssuer(string issuer, SecurityToken securityToken
                 }
             }
 
+            if (configuration != null)
```

Comment:
We should check configuration first, in this path we expect to match configuration.